### ISSUE TITLE
Item Works But Has Issues

### ISSUE THREAD

**Issue by Anston06** [RockinChaos/ItemJoin#346]
I have gotten my item to work([#345](https://github.com/RockinChaos/ItemJoin/issues/345)), but it won't clear when I go to a different world. When I am a normal player(deopped) I can't use the item even thought I have given the itemjoin.ultra and itemjoin.use permissions. When a play's inventory is cleared the item should be given back. Inventory-modify doesn't work either. Anyone know how to fix this?

I would also like it where when the player goes to a different world, it gives them their old item/inventory back. Is this possible?

Config.yml: https://pastebin.com/521dczM5
Items.yml: https://pastebin.com/uiPKmAp0

**Comment by RockinChaos** [comment]
What permissions plugin are you using so I can test with it?

**Comment by Anston06** [comment]
Luckperms

**Comment by Anston06** [comment]
> What permissions plugin are you using so I can test with it?

Luckperms

**Comment by RockinChaos** [comment]
Okay so, I SINCERELY apologize for the huge delay. It took a lot of time for me to track down this bug, it looks like it was a typo on my part.

The latest snapshot will fix this issue; https://ci.craftationgaming.com/job/ItemJoin/591/

Also on the part of returning a cleared player's items, I actually added this as a feature recently, you simply need to add `RETURN_SWITCH` to your Clear Items Options and then specify the worlds you wish you clear in, upon returning to a non-clear world your items will be restored.
Example (in config.yml);
```
Clear-Items:
  Type: ITEMJOIN
  Delay-Tick: 2
  Join: false
  World-Switch: world_the_end, world_nether
  Region-Enter: false
  Options: RETURN_SWITCH
``` 

In this example items are cleared in the nether and the end, so say you are from `world` where the items don't get cleared, those items will be stored when entering those two defined clear-worlds then when you re-enter world your items will be regiven.

I hope that makes sense, it may take a bit of testing for you to get a full understanding of its effect.

**Comment by Anston06** [comment]
Thank you, that will really help me!

**Comment by RockinChaos** [comment]
This bug has been officially fixed in the release of ItemJoin v5.2.0.
You can grab it here; https://www.spigotmc.org/resources/itemjoin.12661/download?version=383801

Please read the changelog as the documentation has not been updated yet;
https://github.com/RockinChaos/ItemJoin/wiki/Recent-Changes
